capstone/src/main/java/com/google/sps/AssociationResult.java

```diff
@@ -77,11 +84,17 @@ public float getWeight() {
     return weight;
   }
 
+  /** @return whether this association has strong sentiment */
+  public boolean hasStrongSentiment() {
+    return strongSentiment;
```

Comment:
Since strong sentiment is computed similarly in each case, should we do

`return Math.abs(this.score) > SENTIMENT_THRESHOLD` instead?

Reply:
In this case I didn't want to mark opinions that were highly divisive as not having a strong sentiment. For example, if one entity had a sentiment of -1 while another had 1 then they could cancel each other out but there are strong feelings there. I therefore check with each new entity added if there is an example of strong sentiment.

Reply:
It seems like strongSentiment doesn't have a very clear definition. Maybe separate out those two different cases. As a developer, I would automatically associated strongSentiment as `return Math.abs(this.score) > SENTIMENT_THRESHOLD` but I might not know that there are edge cases here.

Also, how does entity sentiment cancel each other out? Is that what Sentiment analysis does on their end?

Reply:
Sentiment analysis returns scores of -1 to 1, but if an entity has multiple occurrences with evenly mixed negative and positive scores they can add to something close to 0. CloudNLP gives sentiment per occurrence and I don't believe they provide a way to combine these to get a single score. What I am doing is a weighted sum by the significance of an occurrence in the text (this is given by CloudNLP).

Reply:
Thanks for clarifying this, can you add this as a comment for future readers of the code?